tls_manager_test.go

```diff
@@ -0,0 +1,371 @@
+	certPath, keyPath, expiredCert := writeTestCertFiles(
+		t, true, false, nil,
+	)
+
+	// Now let's run the TLSManager's getConfig. If it works properly, it
+	// should delete the cert and create a new one.
+	cfg := &TLSManagerCfg{
+		TLSCertPath:     certPath,
+		TLSKeyPath:      keyPath,
+		TLSCertDuration: testTLSCertDuration,
+	}
+	tlsManager := NewTLSManager(cfg)
+	_, cleanUp, err := tlsManager.generateOrRenewCert()
+	require.NoError(t, err)
+	_, _, _, err = tlsManager.getConfig()
+	require.NoError(t, err, "couldn't retrieve TLS config")
+	t.Cleanup(cleanUp)
+
+	// Grab the certificate to test that getTLSConfig did its job correctly
+	// and generated a new cert.
+	newCertData, err := tls.LoadX509KeyPair(certPath, keyPath)
+	require.NoError(t, err, "couldn't grab new certificate")
+
+	newCert, err := x509.ParseCertificate(newCertData.Certificate[0])
+	require.NoError(t, err, "couldn't parse new certificate")
+
+	// Check that the expired certificate was successfully deleted and
+	// replaced with a new one.
+	require.True(t, newCert.NotAfter.After(expiredCert.NotAfter),
+		"New certificate expiration is too old")
+}
+
+// Test that the new TLS Manager loads correctly, whether the encrypted TLS key
+// flag is set or not.
+func TestTLSManagerGenCert(t *testing.T) {
+	t.Parallel()
+
+	_, certPath, keyPath := newTestDirectory(t)
+
+	cfg := &TLSManagerCfg{
+		TLSCertPath: certPath,
+		TLSKeyPath:  keyPath,
+	}
+	tlsManager := NewTLSManager(cfg)
+
+	_, _, err := tlsManager.generateOrRenewCert()
+	require.NoError(t, err, "failed to generate new certificate")
+
+	// After this is run, a new certificate should be created and written
+	// to disk. Since the TLSEncryptKey flag isn't set, we should be able
+	// to read it in plaintext from disk.
+	_, keyBytes, err := cert.GetCertBytesFromPath(
+		cfg.TLSCertPath, cfg.TLSKeyPath,
+	)
+	require.NoError(t, err, "unable to load certificate")
+	require.True(t, bytes.HasPrefix(keyBytes, privateKeyPrefix),
+		"key is encrypted, but shouldn't be")
+
+	// Now test that if the TLSEncryptKey flag is set, an encrypted key is
+	// created and written to disk.
+	_, certPath, keyPath = newTestDirectory(t)
+
+	cfg = &TLSManagerCfg{
+		TLSEncryptKey:   true,
+		TLSCertPath:     certPath,
+		TLSKeyPath:      keyPath,
+		TLSCertDuration: testTLSCertDuration,
+	}
+	tlsManager = NewTLSManager(cfg)
+	keyRing := &mock.SecretKeyRing{
+		RootKey: privKey,
+	}
+
+	err = tlsManager.createAndWriteTLSPair(keyRing)
+	require.NoError(t, err, "failed to generate new certificate")
+
+	_, keyBytes, err = cert.GetCertBytesFromPath(
+		certPath, keyPath,
+	)
+	require.NoError(t, err, "unable to load certificate")
+	require.False(t, bytes.HasPrefix(keyBytes, privateKeyPrefix),
+		"key isn't encrypted, but should be")
+}
+
+// TestTLSEncryptDiscrepancies tests a couple of things:
+// 1) If we have cfg.TLSEncryptKey set, but the tls file saved to disk is not
+// encrypted, GenerateOrRenewCert encrypts the file and rewrites it to disk.
```

Comment:
Cool, made these changes. I like the names you suggested. The only one I was iffy on was changing `loadEphemeralCertificate` to `generateEphemeralCertPair`, because I like how the "load" corresponds to `LoadPermanentCertificate`, which does a similar thing. But I changed the rest of them.